lib/ret_web/views/api/v2/credentials_view.ex

```diff
@@ -0,0 +1,42 @@
+defmodule RetWeb.Api.V2.CredentialsView do
+  use RetWeb, :view
+
+  defp render_credentials(credentials) do
+    %{
+      id: credentials.api_credentials_sid,
+      subject_type: credentials.subject_type,
+      is_revoked: credentials.is_revoked,
+      scopes: credentials.scopes,
+      account_id: credentials.account_id,
+      inserted_at: credentials.inserted_at,
+      updated_at: credentials.updated_at,
+      token: nil
+    }
+  end
+
+  def render("index.json", %{credentials: credentials}) when is_list(credentials) do
+    %{
+      credentials: Enum.map(credentials, fn c -> render_credentials(c) end)
+    }
+  end
+
+  def render("show.json", %{token: token, credentials: credentials}) do
+    %{
+      credentials: [Map.merge(render_credentials(credentials), %{token: token})]
+    }
+  end
+
+  def render("show.json", %{credentials: credentials}) do
+    render("show.json", %{credentials: credentials, token: nil})
```

Comment:
Bit strange that `token` is on the outer result here, but inside `credentials:` in the other views.

Reply:
I may be misunderstanding your comment - but `token` always ends up as a key _inside_ the `credentials` json, not outside it. This line is just redirecting to the other `render("show.json", ...)` (which does the call to `render_credentials` and merges in the `token`). 

Open to suggestions to make this clearer -- My intention is to return a `token` in the `credentials` object when you first create one, but then have `token: null` every time after that.

Reply:
Ah, my mistake, I misread the code paths in this case.